test/e2e/storage_test.go

```diff
@@ -0,0 +1,57 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	azureProvider "github.com/openshift/windows-machine-config-operator/test/e2e/providers/azure"
+)
+
+// TestStorage tests that persistent storage can be used with Windows workloads on supported providers
+func TestStorage(t *testing.T) {
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+	// If more cloud provider storage tests are to be added, the code which provisions StorageClasses and PVCs should
+	// be moved into the CloudProvider interface
+	provider, isAzureProvider := tc.CloudProvider.(*azureProvider.Provider)
+	if !isAzureProvider {
+		t.Skip("storage not supported on this platform")
+	}
```

Comment:
With this, if there is an issue with the azure provider that causes the cast to fail, we ignore it and skip storage testing (even if tested on azure). Consider
```
	if tc.CloudProvider.GetType() != config.AzurePlatformType {
		t.Skip("storage not supported on this platform")
	}
	provider, isAzureProvider := tc.CloudProvider.(*azureProvider.Provider)
	require.True(t, isAzureProvider)
```